mkosi/__init__.py

```diff
@@ -312,7 +312,7 @@ def configure_autologin(context: Context) -> None:
 
 @contextlib.contextmanager
 def mount_cache_overlay(context: Context) -> Iterator[None]:
-    if not context.config.incremental or not any(context.root.iterdir()):
+    if not context.config.incremental or not context.config.base_trees or context.config.overlay:
```

Comment:
Just to be entirely clear, since I could read the commit message either way: This is supposed to return early in case there already is an overlay (since there is already one, no need for another) or no base tree is configured (since we don't need an overlay), right?

Reply:
Exactly